### ISSUE TITLE
Run BaseSearchCV instances which are not scikit-learn builtins

### ISSUE THREAD

**Issue by NicolasHug** [openml/openml-sklearn#6]
#### Description

I have a custom implementation of Successive Halving https://github.com/scikit-learn/scikit-learn/pull/13900 which inherits from BaseSearchCV. When I use `run_model_on_task` I'm getting the following warning:

> Warning! Using subclass BaseSearchCV other than {GridSearchCV, RandomizedSearchCV}. Should implement param check. 

I tried to look at the code but couldn't find what those param checks were. It seems that none of the metric are computed on that estimator.

Any help appreciated. Thanks!

#### Versions

OpenML 0.11.0dev
